lib/oci_utils/impl/oci-image-migrate-import.py

```diff
@@ -0,0 +1,240 @@
+    # bucket
+    try:
+        bucketcontents = migrate_utils.bucket_exists(bucket)
+        console_msg(msg='Object storage %s present.' % bucket)
+    except Exception as e:
+        exit_with_msg('Object storage %s not found: %s' % (bucket, str(e)))
+    #
+    # bucket data
+    try:
+        cmd = ['oci', 'os', 'bucket', 'get', '--bucket-name', '%s' % bucket]
+        console_msg(msg='running: %s' % cmd)
+        bucketlist = migrate_tools.run_popen_cmd(cmd)
+        if bucketlist is not None:
+            bucketdict = json.loads(bucketlist)
+        else:
+            console_msg(msg='bucket data for %s is empty.' % bucket)
+    except Exception as e:
+        exit_with_msg('Failed to get data of %s: %s' % (bucket, str(e)))
+    #
+    # bucket contents
+    try:
+        cmd = ['oci', 'os', 'object', 'list', '--bucket-name', '%s' %  bucket]
+        console_msg('running: %s' % cmd)
+        bucketcontents = migrate_tools.run_popen_cmd(cmd)
+        if bucketcontents is not None:
+            bucketcontentsdict = json.loads(bucketcontents)
+        else:
+            console_msg(msg='bucket %s is emtpy.' % bucket)
+        #print('%s' % bucketcontentsdict)
+    except Exception as e:
+        exit_with_msg('Failed to get contents of %s: %s' % (bucket, str(e)))    
+    #
+    # object exists?
+    if migrate_utils.object_exists(bucketcontents, objectname):
+        console_msg(msg='Object %s present in %s.' % (objectname, bucket))
+    else:
+        exit_with_msg('Object %s missing from %s.' % (objectname, bucket))
+    #
+    # display name exists?
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name',
+           '%s' % displayname]
+    objstat = migrate_tools.run_popen_cmd(cmd)
+    if objstat is not None:
+        exit_with_msg('Image with name %s already exists.' % displayname)
+    #
+    # upload?
+    if not read_yn('\n  Uploading %s to %s as %s'
+                   % (objectname, compartment, displayname)):
+        exit_with_msg('Exiting.\n')
+    cmd = ['oci', 'compute', 'image', 'import', 'from-object', '--bucket-name',
+           '%s' % bucket, '--compartment-id', '%s' % compartment_id,
+           '--name', '%s' % objectname, '--namespace', '%s' % namespace,
+           '--launch-mode',  launchmode, '--display-name', '%s' % displayname]
+    console_msg(msg='running: %s' % cmd)
+    try:
+        resval = migrate_tools.run_popen_cmd(cmd)
+        #console_msg(msg='running import: %s' % resval)
+    except Exception as e:
+        exit_with_msg('Failed to import %s: %s' % (objectname, str(e)))
+    #
+    # follow up
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name', '%s' % displayname]
+    # console_msg(msg='running: %s' % cmd)
+    finished = False
+    year, month, day, hour, min = map(int, time.strftime("%Y %m %d %H %M").split())
+    sys.stdout.write('\n  %02d-%02d-%04d %02d:%02d %s\n'
```

Comment:
Why not using your helpers like console_msg ?

Reply:
replaced by better way of indicating activity